airflow/providers/apache/kafka/hooks/admin_client.py

```diff
@@ -0,0 +1,74 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import Any, Sequence
+
+from confluent_kafka.admin import AdminClient, NewTopic
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+
+
+class KafkaAdminClientHook(BaseHook):
+    """KafkaAdminClientHook
+
+    A hook for interacting with the Kafka Cluster
+
+    :param config: A config dictionary to use with confluent_kafka library, defaults to None
+    """
+
+    def __init__(
+        self,
+        config: dict[Any, Any] | None = None,
+    ) -> None:
+        super().__init__()
+
+        self.config: dict[Any, Any] = config or {}
+
+        if not (self.config.get("bootstrap.servers", None)):
+            raise AirflowException("config['bootsrap.servers'] must be provided.")
```

Comment:
WDYT about breaking out `bootstrap_servers` into its own parameter since it's required configuration? Then the `config` param can include _optional_ configuration for the underlying lib.

Reply:
The design principle is to mirror confluent-kafka config as close as possible so that there's no changes taking this to a common consumer or vice versa.